python/docs/lis/userguide.rst

```diff
@@ -56,26 +65,180 @@ also be reached directly from the Logical File:
 
-Read all the curve data by passing each DFSR to :func:`dlisio.lis.curves`:
+.. note::
+    LIS79 opens for the presence of duplicated DFSR, for redundancy. Currently,
+    dlisio has no support for identifying redundant DFSRs, and curves() will
+    return empty arrays for these.
+
+The :class:`dlisio.lis.DataFormatSpec` can be accessed directly from the
+:class:`dlisio.lis.LogicalFile`:
 
 .. code-block:: python
 
-    >>> for format_spec in f.data_format_specs:
-    ...    _ = lis.curves(f, format_spec)
+    >>> formatspecs = f.data_format_specs
+
+A DFRS contains information about the logset in general, such as logging
+direction and information about the index. In addition it contains a list of
+Spec Blocks, :attr:`dlisio.lis.DataFormatSpec.specs`. Each Spec Block describes
+one of the channels/curves in the logset. But let's begin by looking at the
+index:
+
+.. code-block:: python
+
+    >>> format_spec = formatspecs[0]
+    >>> format_spec.index_mnem
+    'TDEP'
+    >>> format_spec.index_units
+    '.1IN'
+    >>> format_spec.spacing
+    60
+    >>> format_spec.spacing_units
+    '.1IN'
+    >>> format_spec.direction
+    255
+
+This tells us that the current logset is indexed against depth, and that the
+depth is measured in 1/10 of an inch. There is also a constant spacing of 60
+.1IN (or 6 IN) between each sample. Note that there is no general requirement
+that the index is evenly spaced and these values might be missing in such
+cases. Lastly we see that the measurements is taken going down-hole (refer to
+:attr:`dlisio.lis.DataFormatSpec.direction`).
+
+Now lets look at the individual channels/curves in the logset. We can list the
+mnemonics and the units of measurement by simply looping the Spec Blocks:
+
+.. code-block:: python
+
+    >>> for spec in format_spec.specs:
+    ...     print("Index: {} (units={})".format(spec.mnem, spec.units))
+    "Index: 'TDEP', units: '.1IN')"
+    "Index: 'TIME', units: 'ms  ')"
+    "Index: 'CHX ', units: 'M   ')"
+    "Index: 'RCCL', units: '    ')"
 
 .. note::
-   LIS79 opens for the presence of duplicated DFSR, for redundancy. Currently,
-   dlisio has no support for identifying redundant DFSRs, and curves() will
-   return empty arrays for these.
+    Please refer to :class:`dlisio.lis.DataFormatSpec` and
+    :class:`dlisio.core.spec_block_0` / :class:`dlisio.core.spec_block_1` for
+    the complete documentation of DFSR and Spec Blocks, respectively.
 
+Reading the curves
+..................
+
+The :class:`dlisio.lis.DataFormatSpec` only contains the metadata about the
+logset. To read the actual curves, pass the DFSR ``format_spec`` to
+:func:`dlisio.lis.curves` together with the filehandle ``f``:
+
+.. code-block:: python
+
+    >>> curves = lis.curves(f, format_spec)
+
+This returns a `structured numpy.ndarray`_ containing all the curves in the
+logset, using the mnemonics from the Spec Blocks as field (column) names.
+
```

Comment:
This returns a ... . Array contains all the curves in the logset and uses the mnemonics...

"using" now refers to wrong word